sql/server.proto

```diff
@@ -32,4 +32,5 @@ message Session {
   optional int32 syntax = 2 [(gogoproto.nullable) = false];
   // Open transaction.
   optional proto.Transaction txn = 3;
+  optional bool mutates_system_db = 4 [(gogoproto.nullable) = false, (gogoproto.customname) = "MutatesSystemDB"];
```

Comment:
could you add a comment? I.e. what keys are those (roughly) and that it's only a valid option when there's a transaction. In fact, to avoid illegal pairings, technically this should be something like

```go
message OpenTransaction {
  required proto.Transaction txn = 1;
  required bool mutates_system_db = 2;
}
optional OpenTransaction open_txn = 3;
```
but we're not using `required` anywhere else.